src/coreclr/src/vm/method.hpp

```diff
@@ -1788,7 +1786,14 @@ class MethodDesc
     //
     PCODE DoBackpatch(MethodTable * pMT, MethodTable * pDispatchingMT, BOOL fFullBackPatch);
 
-    PCODE DoPrestub(MethodTable *pDispatchingMT);
+    // The GC mode for the thread that initially called Prestub().
+    enum
+    {
+        CallerGCMode_Unknown,
+        CallerGCMode_Coop,
+        CallerGCMode_Preempt    // (e.g. NativeCallableAttribute)
+    };
+    PCODE DoPrestub(MethodTable *pDispatchingMT, DWORD callerGCMode = CallerGCMode_Unknown);
```

Comment:
Can this use actual enum name instead of DWORD?

Reply:
I tried for about 20 minutes. It wasn't obvious how to do it if the enum was nested in `MethodDesc` - the type is forward declared all over and the header includes are a giant mess. I could pull out the enum and try putting it in something like `common.h`, but the issue there is naming and there is some annoyance that I didn't look into between `codeman.h` and `codeversion.h`. I can give it another try or if you have a suggestion on where where to attempt that would be appreciated.

Reply:
This entire problem is related to `CodeVersionManager::PublishVersionableCodeIfNecessary()` and the header file declaring that function being included in some very targeted code paths. I can start throwing headers at it to make it build, but that I think that is how we got into the current mess. I need some distance from this. Will pick it up tomorrow.